### PR TITLE
[WIP][DO NOT MERGE] Get rid of Candidate

### PR DESCRIPTION
**Pull request by jrapin**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/base.py [hunk 1]
```diff
@@ -116,17 +59,15 @@ class CandidateMaker:
...
-    def __init__(self, instrumentation: parameter.Instrumentation) -> None:
+    def __init__(self, instrumentation: p.Instrumentation) -> None:
         self._instrumentation = instrumentation
+        self._instrumentation.freeze()
 
-    def __call__(self, args: Tuple[Any, ...], kwargs: Dict[str, Any], data: ArrayLike) -> Candidate:
-        return Candidate(args, kwargs, data)
-
-    def from_call(self, *args: Any, **kwargs: Any) -> Candidate:
+    def fr2om_call(self, *args: Any, **kwargs: Any) -> p.Instrumentation:
```
Comment (danthe3rd) [inline nevergrad/optimization/base.py]: typo I assume?

Comment (jrapin) [reply nevergrad/optimization/base.py]: Not exactly, that was to trigger mypy and tell me if this was still used somewhere -> it's not :D I'll revert indeed though and add deprecation warnings...

Comment (danthe3rd) [reply nevergrad/optimization/base.py]: Oh smart!

File: nevergrad/optimization/base.py [hunk 2]
```diff
@@ -135,15 +76,14 @@ def from_call(self, *args: Any, **kwargs: Any) -> Candidate:
 
         Returns
         -------
-        Candidate:
-            The corresponding candidate. Candidates have field `args` and `kwargs` which can be directly used
+        p.Instrumentation:
+            The corresponding candidate. p.Instrumentations have field `args` and `kwargs` which can be directly used
             on the function (`objective_function(*candidate.args, **candidate.kwargs)`).
         """
```
[the remark this answers is not in the document]
Comment (jrapin) [reply nevergrad/optimization/base.py]: What do you want to update there? There is a whole bunch of doc I need to update indeed, but this is true I think, p.Instrumentation has fields args(=value[0]) and kwargs(=value[1]), it serves as a transition from the old system (and will probably stay there)

**Pull request closed without merging**